source/common/runtime/runtime_impl.cc

```diff
@@ -448,30 +449,103 @@ void ProtoLayer::walkProtoValue(const ProtobufWkt::Value& v, const std::string&
 
 LoaderImpl::LoaderImpl(Event::Dispatcher& dispatcher, ThreadLocal::SlotAllocator& tls,
                        const envoy::config::bootstrap::v2::LayeredRuntime& config,
-                       absl::string_view service_cluster, Stats::Store& store,
-                       RandomGenerator& generator, Api::Api& api)
+                       const LocalInfo::LocalInfo& local_info, Init::Manager& init_manager,
+                       Stats::Store& store, RandomGenerator& generator,
+                       ProtobufMessage::ValidationVisitor& validation_visitor, Api::Api& api)
     : generator_(generator), stats_(generateStats(store)), admin_layer_(stats_),
-      tls_(tls.allocateSlot()), config_(config), service_cluster_(service_cluster), api_(api) {
+      tls_(tls.allocateSlot()), config_(config), service_cluster_(local_info.clusterName()),
+      api_(api) {
   for (const auto& layer : config_.layers()) {
-    if (layer.has_admin_layer()) {
+    switch (layer.layer_specifier_case()) {
+    case envoy::config::bootstrap::v2::RuntimeLayer::kAdminLayer:
       if (config_has_admin_layer_) {
         throw EnvoyException(
             "Too many admin layers specified in LayeredRuntime, at most one may be specified");
       } else {
         config_has_admin_layer_ = true;
       }
-    } else if (layer.has_disk_layer()) {
+      break;
+    case envoy::config::bootstrap::v2::RuntimeLayer::kDiskLayer:
       if (watcher_ == nullptr) {
         watcher_ = dispatcher.createFilesystemWatcher();
       }
       watcher_->addWatch(layer.disk_layer().symlink_root(), Filesystem::Watcher::Events::MovedTo,
                          [this](uint32_t) -> void { loadNewSnapshot(); });
+      break;
+    case envoy::config::bootstrap::v2::RuntimeLayer::kTdsLayer:
+      subscriptions_.emplace_back(
+          std::make_unique<TdsSubscription>(*this, layer.tds_layer(), store, validation_visitor));
+      init_manager.add(subscriptions_.back()->init_target_);
+      break;
+    default:
+      break;
```

Comment:
Should this be NOT_REACHED since we validate in schema?

Reply:
I haven't marked it as `option (validate.required) = true;` in the bootstrap yet; I'm wondering about this in the context of forwards/backwards compat, if an unknown layer is specified, shouldn't we be able to ignore?

Reply:
I guess? Though we don't do this anywhere else AFAIK. Can you at least warn if we decide to go this route?

Reply:
Done.